modules/services/fusuma.nix

```diff
@@ -0,0 +1,108 @@
+{ config, pkgs, lib, ... }:
+let cfg = config.services.fusuma;
+in with lib; {
+  meta.maintainers = with maintainers; [ iosmanthus ];
+  options.services.fusuma = {
+    enable = mkEnableOption
+      "the fusuma systemd service to automatically enable touchpad gesture";
+    package = mkOption {
+      type = types.package;
+      default = pkgs.fusuma;
+      defaultText = literalExpression "pkgs.fusuma";
+      description = "Package providing <command>fusuma</command>.";
+    };
+    settings = mkOption {
+      type = types.attrs;
+      example = literalExpression ''
+        {
+          enable = true;
+          extraPackages = with pkgs;[ xdotool ];
+          config = {
+            threshold = {
+              swipe = 0.1;
+            };
+            interval = {
+              swipe = 0.7;
+            };
+            swipe = {
+              "3" = {
+                left = {
+                  # GNOME: Switch to left workspace
+                  command = "xdotool key ctrl+alt+Right";
+                };
+              };
+            };
+          };
+        };
+      '';
+      description = ''
+        YAML config that will override the default Astroid configuration. 
+      '';
+    };
+    extraPackages = mkOption {
+      type = types.listOf types.package;
+      default = with pkgs; [ coreutils ];
+      defaultText = literalExpression "pkgs.coreutils";
+      example = literalExpression ''
+        with pkgs; [ coreutils xdotool ];
+      '';
+      description = ''
+        Extra packages needs to bring to the scope of fusuma service.
+      '';
+    };
+  };
+  config = let
+    configJSON = pkgs.writeText "config.json" (builtins.toJSON cfg.settings);
```

Comment:
The convention in HM is to use strict camel casing, i.e., `configJson` instead of `configJSON`.